# Task
Calculate the total energy, momentum components, velocity, and Lorentz factor for a muon with the following data: transverse momentum = 54.7055, pseudorapidity = -0.432396, azimuthal angle = 2.574210, and charge.

## Initial data

### Subtask:
Define the given data for the muon (transverse momentum, pseudorapidity, azimuthal angle, and charge).


**Reasoning**:
The subtask requires defining four variables representing the given muon data. This can be done in a single code block.



In [1]:
pt = 54.7055
eta = -0.432396
phi = 2.574210
charge = -1

## Calculate longitudinal momentum

### Subtask:
Calculate the longitudinal momentum using the transverse momentum and pseudorapidity.


**Reasoning**:
Import the numpy library and calculate the longitudinal momentum using the provided formula and variables.



In [2]:
import numpy as np

pz = pt * np.sinh(eta)

## Calculate total momentum

### Subtask:
Calculate the total momentum using the transverse momentum and longitudinal momentum.


**Reasoning**:
Calculate the square of the transverse momentum and the longitudinal momentum, sum them, and then take the square root to find the total momentum, storing the result in the variable `p`.



In [3]:
pt_squared = pt**2
pz_squared = pz**2
sum_of_squares = pt_squared + pz_squared
p = np.sqrt(sum_of_squares)

## Calculate momentum components

### Subtask:
Calculate the x, y, and z components of the momentum using the total momentum, azimuthal angle, and longitudinal momentum.


**Reasoning**:
Calculate the x and y components of the momentum using the total momentum and azimuthal angle, and the z-component is already available.



In [4]:
px = p * np.cos(phi)
py = p * np.sin(phi)
pz = pz

## Calculate total energy

### Subtask:
Calculate the total energy using the total momentum and the known mass of a muon.


**Reasoning**:
Calculate the total energy using the total momentum and the known mass of a muon, following the steps provided in the instructions.



In [5]:
muon_mass = 0.105658  # GeV/c^2
energy = np.sqrt(p**2 + muon_mass**2)

## Calculate velocity

### Subtask:
Calculate the velocity using the total momentum and total energy.


**Reasoning**:
Calculate the velocity using the total momentum and total energy.



In [6]:
velocity = p / energy

## Calculate lorentz factor

### Subtask:
Calculate the Lorentz factor using the velocity.


**Reasoning**:
Calculate the Lorentz factor using the velocity by squaring the velocity, subtracting from 1, taking the square root, and then the reciprocal.



In [7]:
velocity_squared = velocity**2
one_minus_v_squared = 1 - velocity_squared
lorentz_factor = 1 / np.sqrt(one_minus_v_squared)

## Present results

### Subtask:
Display the calculated values for total energy, momentum components, velocity, and Lorentz factor.


**Reasoning**:
Display the calculated values as requested in the subtask instructions.



In [8]:
print(f"Total Energy: {energy}")
print(f"Momentum Components (px, py, pz): ({px}, {py}, {pz})")
print(f"Velocity: {velocity}")
print(f"Lorentz Factor: {lorentz_factor}")

Total Energy: 59.89981337809952
Momentum Components (px, py, pz): (-50.51406261594153, 32.191706341089244, -24.398457919880652)
Velocity: 0.9999984443060068
Lorentz Factor: 566.9217037997913


## Summary:

### Data Analysis Key Findings

*   The total energy of the muon was calculated to be approximately 59.90 GeV.
*   The momentum components were calculated as follows: px $\approx$ -50.51 GeV, py $\approx$ 32.19 GeV, and pz $\approx$ -24.40 GeV.
*   The velocity of the muon was calculated to be approximately 0.999998c.
*   The Lorentz factor was calculated to be approximately 566.92.

### Insights or Next Steps

*   The high Lorentz factor and velocity close to the speed of light indicate that the muon is highly relativistic.
*   These calculated values are essential for further analysis of the muon's behavior and interactions in particle physics experiments.
